custodian/vasp/handlers.py

```diff
@@ -935,62 +934,46 @@ def __init__(self, wall_time=None, buffer_time=300,
                 Should be used with LWAVE = .True. to be useful. If this is
                 True, the STOPCAR is written with LABORT = .TRUE. instead of
                 LSTOP = .TRUE.
-            auto_continue (bool): Use the auto-continue functionality within the
-                VaspJob by ensuring Vasp doesn't delete the STOPCAR
         """
         if wall_time is not None:
             self.wall_time = wall_time
         elif "PBS_WALLTIME" in os.environ:
             self.wall_time = int(os.environ["PBS_WALLTIME"])
         elif "SBATCH_TIMELIMIT" in os.environ:
-            self.walltime = int(os.environ["SBATCH_TIMELIMIT"]) 
+            self.wall_time = int(os.environ["SBATCH_TIMELIMIT"]) 
         else:
             self.wall_time = None
         self.buffer_time = buffer_time
-        self.start_time = datetime.datetime.now()
+        if "JOB_START_TIME" in os.environ:
+            # Enables use of setting JOB_START_TIME as the start
+            # time to use for the walltime handler.  The easiest
+            # way to use this is by executing export JOB_START_TIME=`date`
+            # in the environment (e. g. in a queue submission script)
+            self.start_time = datetime.datetime.strptime(
+                os.environ["JOB_START_TIME"], "%a %b %d %H:%M:%S %Z %Y")
```

Comment:
The problem is if there are multiple custodian runs in the same batch job.  Let's say you run two custodian jobs or a lengthy preprocessing step and a custodian job in the same batch job that both take around 16 hours but you have a 24 hour walltime.  The second will initialize the start_time at the beginning of the *custodian job,* rather than using a time that represents when the batch job began, which was presumably sixteen hours before, and won't trigger the WalltimeHandler appropriately.

It's definitely true that there's a mismatch between when the batch job starts and the custodian job starts, but the point of this is that providing an environment variable from the batch environment is an appropriate reference for the batch walltime limit, rather the one in python that's relative to the custodian job.  It also doesn't change the default behavior (which is to initialize via python), just provides a user with a route to intervene in the aforementioned use case.